File: ch9/yojkim/ch9/dollar.go
package ch9

type Dollar struct {
	amount int
	currency string
}

func NewDollar(amount int) Dollar {
	return Dollar{
		amount: amount,
		currency: "USD",
	}
}

func (d Dollar) Times(multiplier int) Money {
	return NewDollar(d.amount * multiplier)
}

func (d Dollar) Equal(input Money) bool {
	dollar, ok := input.(Dollar)
	if !ok {
		return false
	}
	return d.amount == dollar.amount
}

func (d Dollar) Currency() string {
	return d.currency
}

File: ch9/gunddol/src/ch9/dollar.java
package ch9;

public class dollar extends Money{

    public String currency;

    public dollar(int amount, String currency){
        super(amount, currency);
    }

    Money times(int multiplier) {
        return Money.dollar(amount * multiplier);
    }

//    String currency(){
//        return currency;
//    }
}
